paasd/src/store.rs

```diff
@@ -0,0 +1,97 @@
+use std::{
+    collections::HashMap,
+    sync::{Arc, RwLock},
+};
+
+use uuid::Uuid;
+
+use worker::Process;
+
+use crate::user::UserId;
+
+#[derive(Debug, thiserror::Error)]
+pub enum GetError {
+    #[error("Process with the given id not found")]
+    NotFound,
+    #[error("User unauthorized to access this process")]
+    Unauthorized,
+}
+
+impl Into<tonic::Status> for GetError {
+    fn into(self) -> tonic::Status {
+        use GetError::*;
+        match self {
+            NotFound => tonic::Status::not_found(format!("{}", self)),
+            // TODO: unauthenticated != unauthorized, better Status?
+            Unauthorized => tonic::Status::unauthenticated(format!("{}", self)),
+        }
+    }
+}
+
+pub type ProcessStore = Store<Process>;
+
+type StoreMap<V> = HashMap<Uuid, (UserId, Arc<V>)>;
```

Comment:
The value field would be clearer a full struct and would allow you to eliminate this type alias since the new hashmap type isn't as complex.